Update for Code Structure and Readability

cmchan04 opened the pull request:
Loading csv function has moved to TransactionReader, as so as future loader function into Linked List
Test code will still remain in main.cpp

@BengRhui Oppa please take a look when you are free❤️❤️

cmchan04 committed:
Update for code structure and readability
Loading csv function has moved to TransactionReader, as so as future loader function into linkedlist
Test code will still remain in main.cpp

cmchan04 commented:
PS: not sure if we would include the CMakeList.txt in .gitignore

cmchan04 committed:
Payment Channel Seperator
Seperate transaction records into differen array (as required in question)
Test code by counting to check if any rows are missed out

cmchan04 commented:
PS: I forgot to branch out another commit for Channel seperator, can take a look together 😂😂😂 @BengRhui

cmchan04 commented:
![image](https://github.com/user-attachments/assets/df959242-202c-4317-8145-e530c52e38d1)

BengRhui commented:
> PS: not sure if we would include the CMakeList.txt in .gitignore

I think let's just exclude `CMakeList.txt` as for now, just to make sure there'll be no errors when others are running on their devices.

BengRhui committed:
Modify code formatting + Transaction Constructor

The formatting of code is modified, with additional commands (sorry for being a perfectionist AHAHA). Meanwhile, I have add a constructor for initializing Transaction object to make the code in TransactionReader cleaner >_<

BengRhui committed:
A small mistake here T_T

Didnt thought that removing this is invalid, so its added back AHAHA. For the Transaction struct, I've added curly brackets to let the values initialize by themselves.

BengRhui commented:
@cmchan04 I've added a Transaction.cpp file to include a constructor for Transaction. The other changes is just in terms of formatting, and with comments added for readability.

Whereas for `channelSeparator`, I haven't go thru that yet. But pls take a look at my changes as for now. I think there should be no issue for the rest except `channelSeparator` for now.

BengRhui reviewed (approved):
Everything looks good for this commit, but do let me go through the `channelSeparator` for a while.

cmchan04 reviewed (commented):
LGTM, constructor has simplified our main cpp to make it not too complex, but the removal of TransactionReader:: may cause the code not working becuz we did not initialize any reader object and just use it as const function. Others I think ok

cmchan04 reviewed (commented):
So we assume those value to be 0 and False for boolean when it is empty? maybe we should note down to write in our report assumption, others I think it's fine now.

cmchan04 commented:
![image](https://github.com/user-attachments/assets/e4f0f8f5-bafb-40d1-bf49-d1531d197563)
I did ask sir about the question, so if the channel separator is ok then we have done half of requirement 1 lah, just left creating a linked list and try if the function is applicable on linked list also or not.

BengRhui commented:
> LGTM, constructor has simplified our main cpp to make it not too complex, but the removal of TransactionReader:: may cause the code not working becuz we did not initialize any reader object and just use it as const function. Others I think ok

Yup, fixed that in the latest commit alrdy (if I didnt miss out anything)

BengRhui commented:
> ![image](https://private-user-images.githubusercontent.com/162660825/453936408-e4f0f8f5-bafb-40d1-bf49-d1531d197563.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDk2NDg4ODYsIm5iZiI6MTc0OTY0ODU4NiwicGF0aCI6Ii8xNjI2NjA4MjUvNDUzOTM2NDA4LWU0ZjBmOGY1LWJhZmItNDBkMS1iZjQ5LWQxNTMxZDE5NzU2My5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNjExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDYxMVQxMzI5NDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hMzk2OTk2ODgyNmNiMjBmZWZiN2U4OTE3MjU1NmM5MjA3OGQ4MzBhOTBmMDA1N2U3ZTQxYmJkZmFhYTg0NDJlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.eAHzpjEAzgZpLf8GSM2NPwYAHK5vOYZoc_ijJ-2wQl8) I did ask sir about the question, so if the channel separator is ok then we have done half of requirement 1 lah, just left creating a linked list and try if the function is applicable on linked list also or not.

Ei, then what for we separate 😂 Isn't it useless if we just separate them? N the req is not clear oso, we separate into arrays and linked list or what? 😂 Let's chat in WhatsApp for this, but let me push the last commit from my side first 😂

BengRhui committed:
Tidy up main and Channel Separator

Just modified some comments and commands here

BengRhui commented:
@cmchan04 I've pushed another commit, nothing much is changed except for some comments, the name of the function (coz we split into array mah, not linked list, so it's better to rename it to array), n oso the class name (first letter should be capitalized AHAHA), pls try go thru it haha

But all in all everything LGTM, so I guess u can push to main 👍🏻 Sugoii ne Chun Ming oppa ~
